src/server/serveractiveobject.cpp

```diff
@@ -95,3 +95,9 @@ InventoryLocation ServerActiveObject::getInventoryLocation() const
 {
 	return InventoryLocation();
 }
+
+bool ServerActiveObject::isObservedBy(const std::string &player_name)
+{
+	return !m_observer_names.has_value() ||
+			m_observer_names.value().find(player_name) != m_observer_names.value().end();
+}
```

Comment:
also should probably be inlined
```suggestion
bool ServerActiveObject::isObservedBy(const std::string &player_name) const
{
	return !m_observer_names.has_value() || m_observer_names.value().count(player_name) > 0;
}
```

Reply:
Not sure about inlining. I considered this, but realized I don't have significant grounds to make a call here. See also https://stackoverflow.com/questions/1759300/when-should-i-write-the-keyword-inline-for-a-function-method.